### PR TITLE
Shared Basis Rotate test

### PR DESCRIPTION
**Pull request by garrettwrong**
Replacing two specific (FFB, FLE) basis rotate tests with a simpler (and faster) shared test for all the 2d steerables.

Initially I put them in `test_rotation.py`, migrating that towards pytest.  Changed mind, preferred, `test_rotation.py` to target the `Rotation` class and put the rotate tests in a new file `test_steerable_bases_2d.py`

### CODE REVIEW

File: tests/test_steerable_bases_2d.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    img = gaussian_2d(L, mu=(L // 4, 0), dtype=basis.dtype)
+    img = Image(img / np.linalg.norm(img))  # Normalize
+
+    # Rotate with an ASPIRE steerable basis, returning to real space.
+    rot_img = basis.expand(img).rotate(rot_radians).evaluate()
+
+    # Rotate image with PIL, returning to Numpy array.
+    pil_rot_img = np.asarray(
+        PILImage.fromarray(img.asnumpy()[0]).rotate(
+            rot_radians * 180 / np.pi, resample=PILImage.BILINEAR
```
Comment (j-c-c) [inline tests/test_steerable_bases_2d.py]: Looks like `BICUBIC` gets the `atol` down to .15-ish for the expensive tests.

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks good. Just the one suggestion.

File: tests/test_steerable_bases_2d.py (hunk 1 above)
Comment (garrettwrong) [reply tests/test_steerable_bases_2d.py]: Yeah, that's right, after the normalizing commit things got a little better. Changed.

**Review by j-c-c — APPROVED** [review]
LGTM!

**Comment by garrettwrong** [comment]
Merging.  Centralizes and expands unit test coverage to additional 2D basis. Converts some older tests towards pytest fixtures. No new features.

**Pull request merged**